app.py

```diff
@@ -40,6 +49,118 @@
+        "icon": "BitBucket",  # Replace with the actual icon name or URL
+        "mappings": mappings,
+        "enabled": True,
+        "security": {
+            "secret": WEBHOOK_SECRET,  # Replace with the actual secret for Bitbucket
+            "signatureHeaderName": "X-Hub-Signature",
+            "signatureAlgorithm": "sha256",
+            "signaturePrefix": "sha256=",
+            "requestIdentifierPath": ".headers['X-Request-ID']"  # Optional path based on your needs
+        },
+        "integrationType": "custom"
+    }
+
+    try:
+        response = requests.post(
+            f"{PORT_API_URL}/webhooks",
+            json=webhook_data,
+            headers=port_headers,
+        )
+        response.raise_for_status()
+        webhook_info = response.json()
+        logger.info(f"Successfully created Port webhook: {webhook_info.get('integration', {}).get('url')}")
+        return webhook_info.get('integration', {}).get('url')
+    except requests.exceptions.HTTPError as e:
+        if e.response.status_code == 442:
+            logger.error("Incorrect mapping, kindly fix!")
+            return None
+        logger.error(f"Error creating Port webhook: {e.response.status_code}")
+        return None
+
+
+def get_or_create_project_webhook(project_key: str, webhook_url: str, events: list[str]):
+    logger.info(f"Checking webhooks for project: {project_key}")
+    if webhook_url is not None:
+        try:
+            matching_webhooks = [
+                webhook for project_webhooks_batch in get_paginated_resource(path=f"projects/{project_key}/webhooks")
+                for webhook in project_webhooks_batch if webhook["url"] == webhook_url
+            ]
+            if matching_webhooks:
+                logger.info(f"Webhook already exists for project {project_key}")
+                return matching_webhooks[0]
+            logger.info(f"Webhook not found for project {project_key}. Creating a new one.")
+            return create_project_webhook(project_key=project_key, webhook_url=webhook_url, events=events)
+        except requests.exceptions.HTTPError as e:
+            logger.error(f"HTTP error when checking webhooks for project {project_key}: {e.response.status_code}")
+            return None
+    else:
+        logger.error("Port webhook URL is not available. Skipping webhook check...")
+        return None
+
+
+def create_project_webhook(project_key: str, webhook_url: str, events: list[str]):
+    logger.info(f"Creating webhook for project: {project_key}")
+    webhook_data = {
+        "name": "Port Webhook",
+        "url": webhook_url,
+        "events": events,
+        "active": True,
+        "sslVerificationRequired": False,
+        "configuration": {
+            "secret": WEBHOOK_SECRET,
+            "createdBy": "Port",
+        }
+    }
+    try:
+        response = requests.post(
+            f"{BITBUCKET_API_URL}/rest/api/latest/projects/{project_key}/webhooks",
```

Comment:
any reason why we're using the latest api instead of version 1?

Reply:
Updated, just an oversight